test/torchaudio_unittest/datasets/commonvoice_test.py

```diff
@@ -11,29 +10,49 @@
     normalize_wav,
 )
 
+from torchaudio.datasets import COMMONVOICE
+
 
 class TestCommonVoice(TempDirMixin, TorchaudioTestCase):
```

Comment:
Rather than putting data in class attribute, which makes harder to see the test logic, can you extract them in plain function?

```python
def get_mock_dataset_en(root_dir) -> Tuple[Tensor, int, Dict[str, str]]:
   # create files...

def get_mock_dataset_fr(root_dir) -> Tuple[Tensor, int, Dict[str, str]]:
   # create files...
```

then, split the test classes for different language. Now `setUpClass` should be something as simple as the following.

Also, since we use `wav` format in test, so `CommonVoice._ext_audio` should be changed to `".wav"` before the test.

```python

original_ext_audio = CommonVoice._ext_audio

class TestCommonVoiceEn(TempDirMixin, TorchaudioTestCase):
    backend = 'default'
    root_dir = None
    data = []

@classmethod
def setUpClass(cls):
    cls.root_dir = cls.get_base_temp_dir()
    cls.data = get_mock_dataset_en(cls.root_dir)
    CommonVoice._ext_audio = ".wav"

@classmethod
def tearDownClass(cls):
    CommonVoice._ext_audio = original_ext_audio

def test_commonvoice_str(self)  # this function should stay untouched
    ...

def test_commonvoice_path(self)  # this function should stay untouched
    ...
```

Same goes to French but no need to add `path` version.

Reply:
Hi @mthrok, thanks a lot for your feedback.
I took your comments into consideration. I tried them locally and the tests were green.
I pushed my changes now.
I hope I didn't forget any details.